include/rocksdb/file_checksum.h

```diff
@@ -18,27 +18,42 @@
 
 namespace ROCKSDB_NAMESPACE {
 
-// FileChecksumFunc is the function class to generates the checksum value
+struct FileChecksumGenOptions {
+  std::string file_name;
+  std::string db_name;
+};
+
+// FileChecksumGenerator is the class to generates the checksum value
 // for each file when the file is written to the file system.
-class FileChecksumFunc {
+class FileChecksumGenerator {
  public:
-  virtual ~FileChecksumFunc() {}
-  // Return the checksum of concat (A, data[0,n-1]) where init_checksum is the
-  // returned value of some string A. It is used to maintain the checksum of a
-  // stream of data
-  virtual std::string Extend(const std::string& init_checksum, const char* data,
-                             size_t n) = 0;
+  virtual ~FileChecksumGenerator() {}
 
-  // Return the checksum value of data[0,n-1]
-  virtual std::string Value(const char* data, size_t n) = 0;
+  // Extend the current result after process the data. For different checksum
+  // functions, the temporal results may be stored and used in Extend to
+  // include the new data.
+  virtual void Extend(const char* data, size_t n) = 0;
 
-  // Return a processed value of the checksum for store in somewhere
-  virtual std::string ProcessChecksum(const std::string& checksum) = 0;
+  // Get the checksum
+  virtual std::string GetChecksum() = 0;
 
   // Returns a name that identifies the current file checksum function.
   virtual const char* Name() const = 0;
```

Comment:
Do we need this? Or is it enough to only have it in `FileChecksumGenFactory`?

Reply:
> Do we need this? Or is it enough to only have it in `FileChecksumGenFactory`?

It might be more clear to user different name for FilechecksumGenerator and the Factory.  User can decide to use the same name or not I think.

Reply:
Is it used in code for now? Where do you supposed to use this `Name()`?

Reply:
> Is it used in code for now? Where do you plan to use this `Name()`?

Each time, after the SST file is generated, table builder will call to get the checksum generator name and checksum. The name of ChecksumGenFactory is not used yet.